File: app/src/at-phutran/java/asiantech/internship/summer/fragment/LoginActivity.java
package asiantech.internship.summer.fragment;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import asiantech.internship.summer.R;

public class LoginActivity extends AppCompatActivity {
    private Button mBackButton;
    private TextView mTextViewToolbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
        fragmentTransaction.add(R.id.frContent, new LoginFragment());
        fragmentTransaction.commit();
        mBackButton = findViewById(R.id.btnBack);
        mBackButton.setVisibility(View.GONE);
        backFragment();
    }

    public void onButton() {
        mBackButton.setVisibility(View.VISIBLE);
    }

    public void setTextForToolBar(String title) {
        mTextViewToolbar = findViewById(R.id.tvToolBar);
        mTextViewToolbar.setText(title);
    }

    public void backFragment() {
        mBackButton.setOnClickListener(v -> {
            onBackPressed();
        });
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        Fragment fragment = getSupportFragmentManager().findFragmentById(R.id.frContent);
        if (fragment instanceof LoginFragment) {
            setTextForToolBar(getString(R.string.login));
            mBackButton.setVisibility(View.GONE);
        } else {
            setTextForToolBar(getString(R.string.signUp));
            mBackButton.setVisibility(View.VISIBLE);
        }
    }
}

File: app/src/at-phutran/java/asiantech/internship/summer/kotlin/fragment/LoginActivity.kt
package asiantech.internship.summer.kotlin.fragment

import android.annotation.SuppressLint
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import asiantech.internship.summer.R
import kotlinx.android.synthetic.`at-phutran`.activity_login.*

@SuppressLint("Registered")
class LoginActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        val fragmentTransaction = supportFragmentManager.beginTransaction()
        fragmentTransaction.add(R.id.frContent, LoginFragment())
        fragmentTransaction.commit()
        unShowButtonBack()
        backFragment()
    }

    fun onButton() {
        showButtonBack()
    }

    fun setTextForToolBar(title: String) {
        tvToolBar.text = title
    }

    private fun unShowButtonBack() {
        btnBack.visibility = View.GONE
    }

    private fun showButtonBack() {
        btnBack.visibility = View.VISIBLE
    }

    private fun backFragment() {
        btnBack.setOnClickListener {
            onBackPressed()
        }
    }

    override fun onBackPressed() {
        super.onBackPressed()
        val fragment = supportFragmentManager.findFragmentById(R.id.frContent)
        if (fragment is LoginFragment) {
            tvToolBar.text = getString(R.string.login)
            unShowButtonBack()
        } else {
            tvToolBar.text = getString(R.string.signUp)
            showButtonBack()
        }
    }
}
